consensus/clique/src/main/java/org/hyperledger/besu/consensus/clique/blockcreation/CliqueMiningCoordinator.java

```diff
@@ -33,6 +41,30 @@ public CliqueMiningCoordinator(
     this.miningTracker = miningTracker;
   }
 
+  @Override
+  protected void inSyncChanged(final boolean inSync) {
+    synchronized (this) {
+      final VoteTally validatorProvider =
+          miningTracker
+              .getProtocolContext()
+              .getConsensusState(CliqueContext.class)
+              .getVoteTallyCache()
+              .getVoteTallyAfterBlock(blockchain.getChainHeadHeader());
+      final boolean isValidator =
+          getCoinbase()
+              .filter(coinbase -> validatorProvider.getValidators().contains(coinbase))
+              .isPresent();
+
+      if (inSync && startMiningIfPossible() && isValidator) {
+        LOG.info("Resuming mining operations");
+      }
+
+      if (!inSync && haltCurrentMiningOperation() && isValidator) {
```

Comment:
nit: the haltMiningOperation will stop the current mining operation, which is _sort of_ a side-effect - in the past I've avoided side-effects in conditions - not sure we care.

Reply:
This is no longer present in the CliqueMiningCoordinator class. The original implementation has been moved back to the abstract class. But indeed this behavior was already there